src/main/java/com/powsybl/openloadflow/sensi/AbstractSensitivityAnalysis.java

```diff
@@ -345,6 +404,40 @@ public boolean isConnectedToComponent(Set<LfBus> connectedComponent) {
         }
     }
 
+    static class LfBusVoltagePerTargetV extends LfSensitivityFactor {
+
+        private final LfBus targetVBus;
+
+        LfBusVoltagePerTargetV(SensitivityFactor factor, Network network, LfNetwork lfNetwork, EquationSystem equationSystem) {
+            super(factor, network, lfNetwork, equationSystem);
+            String targetVTerminalId = ((TargetV) factor.getVariable()).getEquipmentId();
+            Terminal regulatingTerminal = getEquipmentRegulatingTerminal(network, targetVTerminalId);
```

Comment:
The factor/bus ref is crappy, this why you are doing all this ugly code. Here factor should provide you a TerminaRef directly.